### PR TITLE
[Work in Progress] - Documentation 

### PR DESCRIPTION
**Pull request by hmanalai**
This PR improves django-declarative-apis documentation by 
1. Modifying the quick start, and resources files. 
2. Adding new files that document django-declarative-apis' core components. 

Test
1. `cd` to the docs folder 
2. Run `make html`
3. Open `index.html` in your browser

Note
The PR still needs approval from @dshafer and @demianbrecht on whether the content of the documentation is correct.

### CODE REVIEW

File: docs/source/quickstart.rst [hunk 1]
```diff
@@ -1,90 +1,193 @@
-Quick start
+Quick Start
```
Comment (briankeane) [inline docs/source/quickstart.rst:1]: This is so, so good.  If I was looking at DDA for the first time I could be kicking tires in about 10 minutes.  Thank you so much for putting such thought into this.

**Review by briankeane — APPROVED** [review]
Really great @hmanalai -- this has been badly needed for so long.

These docs are simple and clear, and they provide a developer-friendly entry-point.   Thanks for the changes!

**Review by demianbrecht — COMMENTED** [review]
This looks great @hmanalai, thanks!

My only real criticism is that it would be preferable to use sphinx [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) when possible, particularly when documenting classes and method parameters. Narratives are great to live outside of the code (or within a module), but class and method documentation is more likely to be updated if it lives alongside the code. Additionally, any new parameters added would automagically be picked up by doc generation rather than relying on someone to manually update it.

File: docs/source/overview.rst [hunk 2]
```diff
@@ -1,12 +1,12 @@
 Overview
 ========
 
-django-declarative-apis is a framework built on top of Django aimed at teams implementing RESTful APis. It provides
+django-declarative-apis is a framework built on top of Django aimed at teams implementing RESTful APIs. It provides
```
Comment (bgrant) [inline docs/source/overview.rst:4]: I wonder why Sphinx isn't rendering the bullets for the list below?

image

File: docs/source/overview.rst [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 a simple interface to define endpoints declaratively. Some benefits to using django-declarative-apis:
 
 * Define endpoints declaratively
 * Define model-bound and unbound resource endpoints with a consistent interface
 * OAuth 1.0a authentication out of the box
 * Define resource and endpoint-bound tasks, promoting modularity
-  * Define synchronous and asynchronous tasks (asynchronous tasks implemented with Celery)
+* Define synchronous and asynchronous tasks (asynchronous tasks implemented with Celery)
```
Comment (bgrant) [inline docs/source/overview.rst:11]: Thanks for fixing this.  It has bugged me for a long time :)

File: docs/source/overview.rst (hunk 2 above)
Comment (hmanalai) [reply docs/source/overview.rst:4]: Yeah, it was working and then suddenly stopped rendering the bullets. It still works with numbered list though.

Comment (bgrant) [reply docs/source/overview.rst:4]: Upgrading the `sphinx_rtd_theme` dev dependency seems to fix it for me.

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -1,90 +1,193 @@
...
+
+        def __str__(self):
+            return f"{self.task}"
+
+
+
+Filters Setup
+---------------
+
+Let’s set up filters in :code:`todo/filters.py` that will decide the format of the api response.
```
Comment (bgrant) [inline docs/source/quickstart.rst:92]: "api" -> "API" to match the usage in overview?

File: docs/source/endpoint_resource.rst [hunk 5]
```diff
@@ -0,0 +1,412 @@
+Endpoint Resource
+==================
+A endpoint resource maps to a single URL endpoint in a Django app’s urls.py. Each endpoint resource can have one or more :code:`EndpointDefinition` that implements the handler code. HTTP verbs such as :code:`POST`, :code:`GET`, along with parameters present in the request determine which :code:`EndpointDefinition` is used.
```
Comment (bgrant) [inline docs/source/endpoint_resource.rst:3]: "A endpoint resource" -> "An endpoint resource"

**Comment by hmanalai** [comment]
@bgrant  I made the suggested changes in the following PR: https://github.com/salesforce/django-declarative-apis/pull/63
This PR uses Sphinx autodoc and it is probably a lot closer to what @demianbrecht  [had suggested](https://github.com/salesforce/django-declarative-apis/pull/62#pullrequestreview-688924414) than this PR. 
I think we can close this one.

**Comment by bgrant** [comment]
> @bgrant I made the suggested changes in the following PR: #63
> PR #63 uses Sphinx autodoc and it is probably a lot closer to what @demianbrecht [had suggested](https://github.com/salesforce/django-declarative-apis/pull/62#pullrequestreview-688924414) than this PR.
> I think we can close this one.

Sounds good @hmanalai !

**Pull request closed without merging**